askama_derive/src/input.rs

```diff
@@ -55,18 +55,20 @@ impl TemplateInput<'_> {
                 PathBuf::from(format!("{}.{}", ast.ident, ext)).into()
             }
             (&Source::Source(_), None) => {
-                return Err("must include 'ext' attribute when using 'source' attribute".into())
+                return Err(CompileError::new(
```

Comment:
I disagree. I removed the `From` implementations on purpose to ensure that we don't use `.into()` by mistake in the future. It's less convenient but it forces to think about whether or not you can provide context on this error or not. If not, then you use `None` and it's done.

Side effect: it also allows us (reviewers) to spot it more easily when reviewing and to check if it could in fact provide context to the error.

Reply:
This change should definitely not be in the same commit as all these other changes, because it is a noop change which makes the diff very verbose and thus hard to review. We can reintroduce it in a later commit, although I would suggest it might make more sense to attach `error()` methods to contexts like `TemplateInput` that can refer to the available context directly.